core/state/snapshot/difflayer.go

```diff
@@ -272,28 +272,24 @@ func (dl *diffLayer) Stale() bool {
 
 // Account directly retrieves the account associated with a particular hash in
 // the snapshot slim data format.
-func (dl *diffLayer) Account(hash common.Hash) (*Account, error) {
+func (dl *diffLayer) Account(hash common.Hash) (*types.StateAccount, error) {
```

Comment:
You have now changed this to return the full `*types.StateAccount`, which, IIUC, is not `slim data format` but the full data format. 
Hence the comment needs to be updated. Also, why did you make this change? Just curious, I have nothing to object right now

Reply:
Honestly it's a very good question. I agree with you, in my PR`Account` API returns full `*types.StateAccount`, whereas `AccountRLP` returns `slim data format`. It's not aligned.

The original motivation is I want to avoid some external conversion from slim to full in stateDB, want to hide the "full-to-slim conversion" notion inside of snapshot.

I will revert this change.